frontend/lib/resolution/Resolver.cpp

```diff
@@ -3335,26 +3335,39 @@ void Resolver::resolveIdentifier(const Identifier* ident) {
       auto inScope = scopeStack.back();
       auto inScopes = CallScopeInfo::forNormalCall(inScope, poiScope);
       auto c = resolveGeneratedCall(context, ident, ci, inScopes);
-      // Ensure we error out for redeclarations within the method itself,
-      // which resolution with an implicit 'this' currently does not catch.
       MatchingIdsWithName redeclarations;
       inScope->lookupInScope(ident->name(), redeclarations, IdAndFlags::Flags(),
                              IdAndFlags::FlagSet());
-      if (!redeclarations.isEmpty()) {
-        LookupConfig config = IDENTIFIER_LOOKUP_CONFIG;
-        if (!resolvingCalledIdent) config |= LOOKUP_INNERMOST;
-        issueAmbiguityErrorIfNeeded(ident, inScope, config);
-      } else {
-        // Save result if successful
-        if (handleResolvedCallWithoutError(result, ident, ci, c) &&
-            emitLookupErrors) {
-          issueErrorForFailedCallResolution(ident, ci, c);
+      if (c.mostSpecific().numBest() == 1) {
+        // Ensure we error out for redeclarations within the method itself,
+        // which resolution with an implicit 'this' currently does not catch.
+        if (!redeclarations.isEmpty()) {
+          auto only = c.mostSpecific().only();
+          bool otherThanParenless = false;
```

Comment:
I don't think it's apparent from reading the code why it's necessarily a parenless method we're considering here -- would suggest adjusting the above comment, and/or renaming the variable to not mention `parenless`.